### ISSUE TITLE
Reader training objective

### ISSUE THREAD

**Issue by hnt4499** [facebookresearch/DPR#149]
Hi,

In the reader training source code, reader's "relevance logits" is computed as:

```python
ignored_index = start_logits.size(1)
start_positions.clamp_(0, ignored_index)
end_positions.clamp_(0, ignored_index)
loss_fct = CrossEntropyLoss(reduce=False, ignore_index=ignored_index)

# compute switch loss
relevance_logits = relevance_logits.view(N, M)
switch_labels = torch.zeros(N, dtype=torch.long).cuda()
switch_loss = torch.sum(loss_fct(relevance_logits, switch_labels))
```

Why are switch labels all zeros? What is the purpose of ignoring index here? (which is equal to sequence length in this case)

**Comment by hnt4499** [comment]
I figured it out: positive passages are always at index 0 of each sample during training.
